### PR TITLE
Migrate CI to GHA

### LINKED ISSUE #17: Migrate CI to GHA
Travis doesn't work anymore, we don't have functional CI right now.

I don't have time to work on this right now.

### PR DESCRIPTION
**Pull request by masatake**
Close #17.
    
Signed-off-by: Masatake YAMATO <yamato@redhat.com>

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      max-parallel: 5
```
Comment (jonashaag) [inline .github/workflows/main.yml]: Do we need this?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      matrix:
+        platform:
+          - ubuntu-latest
+        python-version:
+          - 3.6
```
Comment (jonashaag) [inline .github/workflows/main.yml:17]: I'm in favour of testing old versions as well, to not accidentally drop old versions compatibility.

Comment (jonashaag) [reply .github/workflows/main.yml:17]: And more recent versions :)

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          - 3.7
+          - 3.8
+          - 3.9
+    steps:
+      - uses: actions/checkout@v2
```
Comment (jonashaag) [inline .github/workflows/main.yml]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    strategy:
+      max-parallel: 5
+      matrix:
+        platform:
+          - ubuntu-latest
+        python-version:
+          - 3.6
+          - 3.7
+          - 3.8
+          - 3.9
```
Comment (jonashaag) [inline .github/workflows/main.yml:20]:
```suggestion
          - 3.9
          - 3.10
          - 3.11
```

File: .github/workflows/main.yml (hunk 2 above)
Comment (jonashaag) [inline .github/workflows/main.yml:17]:
```suggestion
          - 2.7
          - 3.4
```

(skipping 3.5 and 3.6 here to keep CI smaller)

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: update package information
+        run: sudo apt-get -y -o APT::Immediate-Configure=false update
```
Comment (jonashaag) [inline .github/workflows/main.yml:28]: What's the purpose of `-o APT::Immediate-Configure=false`?

File: tox.ini [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+python =
+    3.6: py36
+    3.7: py37
+    3.8: py38
+    3.9: py39
+
+[testenv]
+deps = cython
+commands =
+    make
```
Comment (jonashaag) [inline tox.ini:15]: In CI, we should (also?) check that the _committed_ `_readtags.c` file passes the tests. The `make` call regenerates that file and never commits the changes.

**Review by jonashaag — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 5 above)
Comment (masatake) [reply .github/workflows/main.yml:28]: It is for avoiding unwanted interactions.
See also https://superuser.com/questions/199582/apt-error-could-not-perform-immediate-configuration-on.

File: tox.ini (hunk 6 above)
Comment (masatake) [reply tox.ini:15]: `make` is taken from https://github.com/universal-ctags/python-ctags3/blob/master/.travis.yml. I would like to focus on migration. 
Fixing the issue in the original actions run in Travis can be fixed in another pull request.

File: .github/workflows/main.yml (hunk 4 above)
Comment (masatake) [reply .github/workflows/main.yml:20]: As far as trying locally, testing on 3.10 failed.
That on 3.11 may fail, too.
I would like to focus on migration. 
Can I add 3.10 and 3.11 later?

File: .github/workflows/main.yml (hunk 2 above)
Comment (masatake) [reply .github/workflows/main.yml:17]: I don't know well about python's ecosystem.
As far as trying locally, 3.5 is not available.
So I guessed ones older than 3.5 were not available.
How about the other versions later?

Comment (sbraz) [reply .github/workflows/main.yml:17]: IMO users of Python < 3.6 can live with the old version of the package. We should just drop compatibility for them. Even Ubuntu 18.04 which is 3 years old has 3.6: https://packages.ubuntu.com/bionic/python3

File: .github/workflows/main.yml (hunk 4 above)
Comment (jonashaag) [reply .github/workflows/main.yml:20]: Sure!

File: .github/workflows/main.yml (hunk 5 above)
Comment (jonashaag) [reply .github/workflows/main.yml:28]: Ok, I have never seen this before but fine with me

File: .github/workflows/main.yml (hunk 2 above)
Comment (jonashaag) [reply .github/workflows/main.yml:17]: If it doesn’t require a lot of effort IMO we should keep it compatible with older versions. Maybe we can stick to whatever recent versions of Cython support.

Comment (jonashaag) [reply .github/workflows/main.yml:17]: Those should be available from the GitHub action.

File: tox.ini (hunk 6 above)
Comment (jonashaag) [reply tox.ini:15]: Oh, agreed.

**Review by jonashaag — APPROVED** [review]

File: .github/workflows/main.yml (hunk 2 above)
Comment (sbraz) [reply .github/workflows/main.yml:17]: > Maybe we can stick to whatever recent versions of Cython support.

Sure, well [this means 3.7 and newer](https://endoflife.date/python), that's why I thought the current file with 3.6+ wasn't bad.

Comment (jonashaag) [reply .github/workflows/main.yml:17]: Oh ok!

Comment (jonashaag) [reply .github/workflows/main.yml:17]: Does 3.6 even build then?

Comment (jonashaag) [reply .github/workflows/main.yml:17]: Oh no I meant what’s the oldest version of Python that is supported by the latest stable release of Cython

Comment (jonashaag) [reply .github/workflows/main.yml:17]: Seems to be 2.6, 2.7, 3.3+

Comment (sbraz) [reply .github/workflows/main.yml:17]: I read that as C*p*ython, sorry!

File: .github/workflows/main.yml (hunk 1 above)
Comment (masatake) [reply .github/workflows/main.yml]: Fixed.

File: .github/workflows/main.yml (hunk 5 above)
Comment (masatake) [reply .github/workflows/main.yml:28]: I added comments about the option.

**Comment by masatake** [comment]
I updated the pull request.
About the versions of python used in test, I specified what I can test locally because I don't know it well. I would like you to adjust the version after merging this pull request.

I ignored "testing `committed _readtags.c`".
It should be done in another pull request, "fixing the way to test in make".

**Comment by masatake** [comment]
@jonashaag, I re-request you review this.

**Review by jonashaag — APPROVED** [review]

**Pull request merged**